drivers/gpio/gpio_smartbond.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2022, Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_smartbond_gpio
+
+#include <stdint.h>
+#include <drivers/gpio.h>
+#include <DA1469xAB.h>
+#include "gpio_utils.h"
+
+/* GPIO P0 and P1 share single GPIO and WKUP peripheral instance with separate
+ * set registers for P0 and P1 interleaved. The starting registers for direct
+ * data access, bit access, mode, latch and wake-up controller are defined in
+ * device tree.
+ */
+
+struct gpio_smartbond_data_regs {
+	uint32_t data;
+	uint32_t _reserved0;
+	uint32_t set;
+	uint32_t _reserved1;
+	uint32_t reset;
+};
+
+struct gpio_smartbond_latch_regs {
+	uint32_t latch;
+	uint32_t set;
+	uint32_t reset;
+};
+
+struct gpio_smartbond_wkup_regs {
+	uint32_t select;
+	uint32_t _reserved0[4];
+	uint32_t pol;
+	uint32_t _reserved1[4];
+	uint32_t status;
+	uint32_t _reserved2[2];
+	uint32_t clear;
+	uint32_t _reserved3[2];
+	uint32_t sel;
+};
+
+struct gpio_smartbond_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	sys_slist_t callbacks;
+};
+
+struct gpio_smartbond_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	uint8_t port;
+	volatile struct gpio_smartbond_data_regs *data_regs;
+	volatile uint32_t *mode_regs;
+	volatile struct gpio_smartbond_latch_regs *latch_regs;
+	volatile struct gpio_smartbond_wkup_regs *wkup_regs;
+};
+
+static void gpio_smartbond_wkup_init(void)
+{
+	static bool wkup_init;
+
+	/* Wakeup controller is shared for both GPIO ports and should
+	 * be initialized only once.
+	 */
+	if (!wkup_init) {
+		WAKEUP->WKUP_CTRL_REG = 0;
+		WAKEUP->WKUP_CLEAR_P0_REG = 0xffffffff;
+		WAKEUP->WKUP_CLEAR_P1_REG = 0xffffffff;
+		WAKEUP->WKUP_SELECT_P0_REG = 0;
+		WAKEUP->WKUP_SELECT_P1_REG = 0;
+		WAKEUP->WKUP_SEL_GPIO_P0_REG = 0;
+		WAKEUP->WKUP_SEL_GPIO_P1_REG = 0;
+		WAKEUP->WKUP_RESET_IRQ_REG = 0;
+
+		CRG_TOP->CLK_TMR_REG |= CRG_TOP_CLK_TMR_REG_WAKEUPCT_ENABLE_Msk;
+
+		WAKEUP->WKUP_CTRL_REG = WAKEUP_WKUP_CTRL_REG_WKUP_ENABLE_IRQ_Msk;
+
+		wkup_init = true;
+	}
+}
+
+static int gpio_smartbond_pin_configure(const struct device *dev,
+				      gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_smartbond_config *config = dev->config;
+
+	if (flags == GPIO_DISCONNECTED) {
+		return 0;
+	}
```

Comment:
this option should actually disable/disconnect the gpio (if applicable, e.g. to save power)